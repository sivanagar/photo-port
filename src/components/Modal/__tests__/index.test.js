import React from 'react';
import { render, cleanup , fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';


const currentPhoto = {
  name: 'Park bench',
  category: 'landscape',
  description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie',
  index: 1
};

const mockToggleModal = jest.fn();

afterEach(cleanup);

describe('Modal Component', () => {
  it('renders', () => {
    render(<Modal currentPhoto={currentPhoto} />)
  });
  it('matches snapshot DOM node stracture', ()=> {
    const { asFragment } = render(<Modal currentPhoto={currentPhoto} />);
    expect(asFragment()).toMatchSnapshot();
  });
  describe('Click Event', () => {
    it('calls onClose handler', () => {
      // Arrange: Render Modal
      const { getByText } = render(<Modal
        onClose={mockToggleModal}
        currentPhoto={currentPhoto}
      />);
      // Act: Simulate click event
      fireEvent.click(getByText('Close this modal'));
      // Assert: Expected matcher
      expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
  })  
})
