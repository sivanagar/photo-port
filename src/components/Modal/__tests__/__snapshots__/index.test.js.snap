// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component matches snapshot DOM node stracture 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park bench
      </h3>
      <img
        alt="current category"
      />
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc ultricie
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
